src/vs/platform/extensionManagement/common/extensionManagement.ts

```diff
@@ -85,6 +91,7 @@ export interface IExtensionContributions {
 	menus?: { [context: string]: IMenu[] };
 	snippets?: ISnippet[];
 	themes?: ITheme[];
+	explorer?: ITreeExplorer;
```

Comment:
Decided to start with this, because I don't know which path we'll take if we were to allow more complex explorer contribution in the future.

1. Allow multiple sections (tree or non-tree) like debug

  ![image](https://cloud.githubusercontent.com/assets/4033249/19822810/3c78dcd8-9d1a-11e6-8061-4d029cf16499.png)

2. Allow multiple explorers per extenseion

Better to keep it minimal for first iteration I think.